codebuild/spec/buildspec_valgrind.yml

```diff
@@ -0,0 +1,54 @@
+---
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+version: 0.2
+
+batch:
+  build-list:
+    - identifier: gcc_awslc
+      env:
+        compute-type: BUILD_GENERAL1_LARGE
+        variables:
+          S2N_LIBCRYPTO: awslc
+          COMPILER: gcc
+    - identifier: gcc_openssl_3_0
+      env:
+        compute-type: BUILD_GENERAL1_LARGE
+        variables:
+          S2N_LIBCRYPTO: openssl-3.0
+          COMPILER: gcc
+
+phases:
+  pre_build:
+    commands:
+      - |
+        if [ -d "third-party-src" ]; then
+          cd third-party-src;
+        fi
+      - /usr/bin/$COMPILER --version
+  build:
+    on-failure: ABORT
+    commands:
+      - |
+        cmake . -Bbuild \
+          -DCMAKE_C_COMPILER=/usr/bin/$COMPILER \
+          -DCMAKE_PREFIX_PATH=/usr/local/$S2N_LIBCRYPTO \
+          -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - cmake --build ./build -- -j $(nproc)
+  post_build:
+    on-failure: ABORT
+    commands:
+      - |
+        S2N_VALGRIND=1 CTEST_PARALLEL_LEVEL=$(nproc) \
+          CTEST_OUTPUT_ON_FAILURE=1 ctest -T memcheck \
+          --test-dir build ARGS="-output-on-failure"
```

Comment:
These CI blocks quickly become very difficult to read, so I actually like to keep things on their own line. I'm also assuming that we don't need both `CTEST_OUTPUT_ON_FAILURE=1` and `ARGS="-output-on-failure"`?
```suggestion
        S2N_VALGRIND=1 \
        CTEST_PARALLEL_LEVEL=$(nproc) \
        CTEST_OUTPUT_ON_FAILURE=1 \
        ctest -T memcheck \
              --test-dir build
```

Reply:
I have updated this part. It will be reflected in the next commit.